server/src/main/java/org/elasticsearch/search/SearchException.java

```diff
@@ -51,4 +52,15 @@ protected void writeTo(StreamOutput out, Writer<Throwable> nestedExceptionsWrite
     public SearchShardTarget shard() {
         return this.shardTarget;
     }
+
+    @Override
+    public RestStatus status() {
+        if (getCause() == null) {
```

Comment:
This is a good assumption for the current usages, I wonder if it holds in the future. It is true that whenever you don't have a cause, it means you don't have an internal exception and that's likely a user error. I am a bit torn on whether the fix can be so simple though. I must admit that even the idea of a SearchException raises doubts for me on when this exception should be used.

Reply:
Agreed. I think in this case it also makes sense to create a new exception as here: https://github.com/elastic/elasticsearch/pull/104868

Most of the cases where SearchException is created without a cause due to a user / config error are in `SearchServer.parseSource`. There are 14 total places where it is created without a cause, so it will be straightforward to have them throw something like `SearchUsageException`.

If we need to return a different status for this scenario vs. other scenarios with SearchException, I see only three options:

1) create a new Exception
2) add an additional field to SearchException, such as `boolean inputValidationFailure` and set status based on that
3) parse the error message text to determine/guess how to set status

The first two both require a new TransportVersion. The latter is very fragile.

So I would vote for number 1, as I think exception classes are should be specific, not general.